CI: Runtime test (privileged) TestNodesPodCIDRManager_Resync

```
 FAIL  package: github.com/cilium/cilium/pkg/ipam/allocator/podcidr   ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛

--- FAIL: TestNodesPodCIDRManager_Resync (0.00s)

    podcidr_test.go:286: 
        	Error Trace:	/host/pkg/ipam/allocator/podcidr/podcidr_test.go:286
        	            				/host/pkg/ipam/allocator/podcidr/podcidr_test.go:300
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestNodesPodCIDRManager_Resync

```

https://github.com/cilium/cilium/actions/runs/14202599380/job/39792877246

[cilium-junits.zip](https://github.com/user-attachments/files/19556370/cilium-junits.zip)

--- a/pkg/ipam/allocator/podcidr/podcidr_test.go
+++ b/pkg/ipam/allocator/podcidr/podcidr_test.go
@@ -281,7 +281,10 @@ func TestNodesPodCIDRManager_Resync(t *testing.T) {
 				}
 			},
 			testPostRun: func(fields *fields) {
-				time.Sleep(2 * time.Millisecond)
+				// Trigger is async, so until we have synctest testing we have
+				// to resort to Eventually.
+				require.Eventually(t, func() bool { return reSyncCalls.Load() >= 1 },
+					time.Second*2, time.Millisecond)
 				require.Equal(t, int32(1), reSyncCalls.Load())
 			},
 		},

